sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -564,13 +565,18 @@ public IterableStream<ServiceBusReceivedMessage> receiveMessages(int maxMessages
         // Since the subscriptions may happen at different times, we want to replay results to downstream subscribers.
         final Sinks.Many<ServiceBusReceivedMessage> emitter = Sinks.many().replay().all();
 
-        queueWork(maxMessages, maxWaitTime, emitter);
-
-        final Flux<ServiceBusReceivedMessage> messagesFlux =  tracer.traceSyncReceive("ServiceBus.receiveMessages", emitter.asFlux());
-        // messagesFlux is already a hot publisher, so it's ok to subscribe
-        messagesFlux.subscribe();
+        final Runnable processWorkQueue = queueWork(maxMessages, maxWaitTime, emitter);
 
-        return new IterableStream<>(messagesFlux);
+        final Flux<ServiceBusReceivedMessage> messages =  tracer.traceSyncReceive("ServiceBus.receiveMessages", emitter.asFlux());
+        // The messages Flux is already a hot publisher, so it's ok to subscribe
+        messages.subscribe(__ -> { },
+            __ -> {
+                Schedulers.boundedElastic().schedule(processWorkQueue);
+            },
+            () -> {
+                Schedulers.boundedElastic().schedule(processWorkQueue);
+            });
```

Comment:
as noted in the description, once the current work signals termination, attempt the work-queue processing.